pkg/gomod/check.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gomod
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver/v4"
+
+	"knative.dev/test-infra/pkg/git"
+)
+
+// Check examines a go mod file for dependencies and  determines if each have a release artifact
+// based on the ruleset provided.
+//
+// See Also
+//
+// Check leverages the same rules used by
+// knative.dev/test-infra/pkg/git.Repo().BestRefFor
+func Check(gomod, release, domain string, ruleset git.RulesetType, verbose bool) error {
+	modulePkgs, _, err := Modules([]string{gomod}, domain)
+	if err != nil {
+		return err
+	}
+
+	for module, packages := range modulePkgs {
+		if err := check(module, packages, release, ruleset, verbose); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func check(module string, packages []string, release string, ruleset git.RulesetType, verbose bool) error {
+	this, err := semver.ParseTolerant(release)
+	if err != nil {
+		return err
+	}
+
+	if verbose {
+		fmt.Printf("%s\n", module)
+	}
```

Comment:
if debug remove
if logging, let's use `log.Println`
By default it goes to stderr, so you'll see on screen

Reply:
I don't want it to go to stderr

Reply:
I don't want `log`, I want `fmt`. Log adds extra stuff to the prefix

Reply:
good call, moved to a `io.Writer` and the caller can choose.